modules/functional_expansion_tools/src/series/Legendre.C

```diff
@@ -17,9 +18,36 @@
 
 Legendre::Legendre() : SingleSeriesBasisInterface() {}
 
-Legendre::Legendre(const std::vector<MooseEnum> & domain, const std::vector<std::size_t> & order)
+Legendre::Legendre(const std::vector<MooseEnum> & domain,
+                   const std::vector<std::size_t> & order,
+                   MooseEnum expansion_type,
+                   MooseEnum generation_type)
   : SingleSeriesBasisInterface(domain, order, calculatedNumberOfTermsBasedOnOrder(order))
 {
+  switch (expansion_type)
+  {
+    case ORTHONORMAL:
+      evaluateExpansionWrapper = std::bind(&Legendre::evaluateOrthonormal, this);
+      break;
+    case SQRT_MU:
+      evaluateExpansionWrapper = std::bind(&Legendre::evaluateSqrtMu, this);
+      break;
+    case STANDARD:
+      evaluateExpansionWrapper = std::bind(&Legendre::evaluateStandard, this);
+      break;
+  }
+  switch (generation_type)
+  {
+    case ORTHONORMAL:
+      evaluateGenerationWrapper = std::bind(&Legendre::evaluateOrthonormal, this);
+      break;
+    case SQRT_MU:
+      evaluateGenerationWrapper = std::bind(&Legendre::evaluateSqrtMu, this);
+      break;
+    case STANDARD:
+      evaluateGenerationWrapper = std::bind(&Legendre::evaluateStandard, this);
+      break;
+  }
```

Comment:
I really like how clean this implementation is.

Reply:
Using a lambda here is another option that some might find cleaner. This is fine though, no change necessary.